edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/IConfigSource.cs

```diff
@@ -5,12 +5,20 @@ namespace Microsoft.Azure.Devices.Edge.Hub.Core.Config
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
 
+    /// <summary>
+    /// Represents a config source for EdgeHubConfig.
+    /// Used to abstract in-memory config source vs twin config source.
+    /// </summary>
     public interface IConfigSource
     {
         Task<Option<EdgeHubConfig>> GetCachedConfig();
 
         Task<Option<EdgeHubConfig>> GetConfig();
 
-        void SetConfigUpdatedCallback(Func<EdgeHubConfig, Task> callback);
+        /// <summary>
+        /// Occurres when config is updated in the config source.
+        /// For example, by a twin update pushed from the cloud.
+        /// </summary>
+        event EventHandler<EdgeHubConfig> ConfigUpdated;
```

Comment:
Why change this to an event?

Reply:
B/c now I have multiple (two) subscribers: `ConfigUpdater` and `PolicyUpdateHandler`.

Reply:
You can still manage multiple callback in the ConfigSource implementation.  For example, you can define same event inside the ConfigSource implementation and calling SetConfigUpdatedCallback (Maybe rename it to AddConfigUpdatedCallback) will add callback to the event.

Reply:
I think event approach is fine here.